hudi-common/src/main/java/org/apache/hudi/common/model/HoodieCommitMetadata.java

```diff
@@ -106,6 +150,14 @@ public void setCompacted(Boolean compacted) {
     return filePaths;
   }
 
+  public void setOperateType(Type type) {
```

Comment:
We also need to add the enum type to the avro schema hudi-common/src/main/avro/HoodieCommitMetadata.avsc

For archiving, we use the avro class org.apache.hudi.avro.model.HoodieCommitMetadata instead of org.apache.hudi.common.model.HoodieCommitMetadata

HoodieCommitArchiveLog.commitMetadataConverter needs to change to copy the operation types to avro objects.